### PR TITLE
Extends docs stimulus contributors

### PR DESCRIPTION
**Pull request by Lovelyfin00**
Based off #10127 

### Extends off the PR by Thibaud on Stimulus contributors guide

Added more content with mine and @lb- reviews.

### CODE REVIEW

**Comment by salty-ivy** [comment]
i would definitely like to see this to extend up to point where we get some idea for writing ### tests for these controllers as well,  because they could be really tricky sometimes. and as far as i am aware there's no docs present for those at all.

**Comment by salty-ivy** [comment]
additionally, in Docs when describing about python tests. there's  a special section explaining how to run individual python tests
https://docs.wagtail.org/en/stable/contributing/developing.html#running-only-some-of-the-tests

we should also add that same symmetrical sections for `npm run test:unit` describing how to run tests individually for stimulus controllers and rest. describing command `npm run test:unit DemoController.test.js`.

about here: https://docs.wagtail.org/en/stable/contributing/developing.html#unit-tests-for-javascript

Cc : @lb- @Lovelyfin00.

File: docs/contributing/ui_guidelines.md [hunk 1]
```diff
@@ -42,6 +43,36 @@ We use [Prettier](https://prettier.io/) for formatting and [ESLint](https://esli
...
+### How to build a Stimulus controller
+
+First think of how to name the controller. Keep it concise, one or two words ideally. Then,
+
+1. Start with the HTML templates, build as much of the UI as you can in HTML alone. Ensure it is accessible and follows the CSS guidelines.
+2. Create the controller file in our `client/src/controllers` folder, along with its tests (see [](testing)) and Storybook stories.
+3. For initialisation, consider which [controller lifecycle methods](https://stimulus.hotwired.dev/reference/lifecycle-callbacks#methods) to use, if any (`connect`, `initialize`).
+4. If relevant, also consider how to handle the controlled element being removed from the DOM [`disconnect` lifecycle method](https://stimulus.hotwired.dev/reference/lifecycle-callbacks#disconnection). 
+5. Document controller classes and methods with [JSDoc annotations](https://jsdoc.app/index.html).
+6. Arrange the `data attributes` in a consistent way. The `id`, `class`, and `type` comes first followed by `data-controller` and then alpha sorting of all other data attributes.
```
Comment (lb-) [inline docs/contributing/ui_guidelines.md:73]:
```suggestion
6. Use [values](https://stimulus.hotwired.dev/reference/values) to provide options and also state. Only reach for instance properties for more complex internal state such as references to resolved DOM elements or other instances. Try to have falsey defaults and avoid too much usage of the Object type when using values.
```

I think we already cover the DOM attributes ordering above, but here is a replacement item 6 that may be useful.

**Comment by lb-** [comment]
Changes have been merged back into https://github.com/wagtail/wagtail/pull/10127

**Pull request closed without merging**